en/developer_guide/esi.rst

```diff
@@ -0,0 +1,92 @@
+ESI blocks
+==========
+
+Open Orchestra leverages the power of ESI blocks to optimize page rendering and increase performance.
+ESI blocks are a way to split the rendering of a page into different elements
+that will be rendered in separate requests by the web server.
+This behavior allows the application to apply HTTP cache to some parts of the page
+and refresh only the outdated parts when needed.
+
+In the FrontOffice application, each block is rendered in an ESI block using the render_esi() twig function.
+The blocks will therefore be cached by an HTTP cache unless
+the method ``DisplayBlockInterface::isPublic()`` returns false.
+
+Setup
+-----
+
+In order to use the power of ESI blocks, it's needed to have a reverse proxy
+in front of the web application that will be able to process them.
+Examples of reverse proxies are Varnish or the HttpCache component of Symfony.
+
+If there is no ESI-capable reverse proxy configured,
+Symfony will still be able to render your pages with just HTML so the application won't suffer any drawbacks.
+
+HttpCache configuration
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The configuration of Symfony for using HttpCache is explained in the `Http cache documentation`_
+
+
+Varnish configuration
+~~~~~~~~~~~~~~~~~~~~~
+
+To let Symfony know that there is an ESI-capable server (Varnish) installed in front of the PHP webserver,
+this ESI server should set an HTTP header in the request to be passed to the application,
+and parse the response to check for ESI usage.
+
+This is done inside the varnish configuration file.
+
+.. code-block:: varnish4
+
+    sub vcl_recv {
+        set req.http.Surrogate-Capability = "varnish=ESI/1.0";
+    }
+
+    sub vcl_fetch {
+        if (beresp.ttl > 0s) {
+            unset beresp.http.Set-Cookie;
+            return(deliver);
+        }
+    }
+
+The power of tags
+-----------------
+
+**This feature is not available with the Symfony HttpCache component.**
+
+Principle
+~~~~~~~~~
+
+The main interest when using ESI tags is to have some parts of a cached page
```

Comment:
The main interest of ESI tags usage is to give the possibility to flush the cache of one or more request based on their content